client/app.js

```diff
@@ -1,17 +1,148 @@
-const homePage = {template: `
-  <div>home</div>
-`};
-const bookPage = {template: `
-  <div>book {{ $route.params.id }}</div>
-`};
+import { getBooks, getBook, setChapterRead, setChapterUnread, createBook } from './api.js';
+
+const homePage = {
+  template: `
+    <div>
+      <router-link to="/add">Add a book</router-link>
+      <ul v-if="books">
+        <li v-for="book in books" :key="book.id">
+          <router-link :to="'/book/' + book.id">
+            {{ book.title }}
+          </router-link>
+        </li>
+      </ul>
+      <ul v-else>
+        <li>...</li>
+      </ul>
+    </div>
+  `,
+  data: function () {
+    return {
+      books: undefined
+    };
+  },
+  created: function () {
+    getBooks()
+      .then(books => {
+        this.books = books;
+      });
+  },
+};
+const bookPage = {
+  template: `
+    <div>
+      <router-link to="/">Home</router-link>
+      <ul v-if="chapters">
+        <li v-for="chapter in chapters" :key="chapter.id">
+          <my-checkbox
+            :read="!!chapter.read_at"
+            :disabled="loading[chapter.id]"
+            @toggle="toggle(chapter)"
+          ></my-checkbox>
+          {{ chapter.title }}
+        </li>
+      </ul>
+      <ul v-else>
+        <li>...</li>
+      </ul>
+    </div>
+  `,
+  data: function () {
+    return {
+      chapters: undefined,
+      loading: undefined,
+    };
+  },
+  created: function () {
+    this.load();
+  },
+  methods: {
+    load: function () {
+    const bookId = this.$route.params.id;
+      getBook(bookId)
+        .then(book => {
+          this.chapters = book.chapters;
+          this.loading = Object.fromEntries(
+            this.chapters.map(
+              chapter => [chapter.id, false]
+            )
+          );
+        });
+    },
+    toggle: async function (chapter) {
+      const setChapterState = chapter.read_at ? setChapterUnread : setChapterRead;
+      this.loading[chapter.id] = true;
+      await setChapterState(chapter.id);
+      await this.load();
+      this.loading[chapter.id] = false;
```

Comment:
❓ Maybe after this PR, try just minimally updating some state (stored in the my-checkbox?) instead of reloading

Reply:
probably using the "initial state" pattern here https://vuejs.org/v2/guide/components-props.html#One-Way-Data-Flow